vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/ComboBox.java

```diff
@@ -938,10 +936,11 @@ private void confirmUpdate(int id) {
 
     @ClientCallable
     private void setRequestedRange(int start, int length, String filter) {
-        this.lastFilter = filter;
-
         dataCommunicator.setRequestedRange(start, length);
-        filterSlot.accept(filter);
+        if (!Objects.equals(filter, lastFilter)) {
```

Comment:
Probably it shouldn't be checked over here. Instead the logic should be changed when setting dataProvider for `filterSlot`.

Reply:
Done